stellar-contract-env-host/src/test.rs

```diff
@@ -1,17 +1,58 @@
+use stellar_contract_env_common::RawValConvertible;
+
 use crate::{
     xdr::{ScObject, ScObjectType, ScVal, ScVec},
-    Host, IntoEnvVal, Object,
+    Host, IntoEnvVal, Object, Tag,
 };
 
 #[test]
-fn i64_roundtrip() {
+fn u64_roundtrip() {
+    let host = Host::default();
+    let u: u64 = 38473_u64; // This will be treated as a ScVal::I64
```

Comment:
This line: https://github.com/stellar/rs-stellar-contract-env/blob/main/stellar-contract-env-common/src/env_val.rs#L208
If the u64 can be fit into i64, then it is unchecked from i64. 
If not, which the test code right after this block checks, it will create an Object::U64 that stores the u64. 
In both cases, it is able to be converted back to u64 via `try_from`.  Does that answer your question?

Reply:
This reminded me to add another case to test negative i64, which will be stored as Object::I64.

Reply:
Ah, that's not good. I removed that logic, and somehow it came back. I'll open a PR removing it.

Reply:
Following this up here: https://github.com/stellar/rs-stellar-contract-env/pull/79